docs/user-guide/configure.md

```diff
@@ -1,53 +1,32 @@
 # Configuring
 
-Stylelint expects a configuration object, and looks for one in a:
+Stylelint expects a configuration object.
 
-- `stylelint.config.js` or `.stylelintrc.js` file
-  - Which module system to use depends on your [default module system configuration](https://nodejs.org/api/packages.html#determining-module-system) for Node.js (e.g., `"type": "module"` in `package.json`).
-- `stylelint.config.mjs` or `.stylelintrc.mjs` file using `export default` (ES module)
-- `stylelint.config.cjs` or `.stylelintrc.cjs` file using `module.exports` (CommonJS)
-- `.stylelintrc.json`, `.stylelintrc.yml`, or `.stylelintrc.yaml` file
-- `.stylelintrc` file in JSON or YAML format
-  - We recommend adding an extension (e.g., `.json`) to help your editor provide syntax checking and highlighting.
-- `stylelint` property in `package.json`
-
-ES module example:
-
-```js
-/** @type {import('stylelint').Config} */
-export default {
-  rules: {
-    "block-no-empty": true
-  }
-};
-```
+Starting from the current working directory, Stylelint searches upwards until it finds a `stylelint.config.{js,mjs}` file that exports one. Or, you can use the [`--config`](cli.md#--config--c) or [`configFile`](options.md#configfile) option to short-circuit the search.
 
 > [!NOTE]
-> The `@type` JSDoc annotation enables Typescript to autocomplete and type-check.
+> Stylelint currently supports other configuration formats, but we may remove these in the future:
```

Comment:
As this list also includes `.mjs` and `.js` it isn't clear which format is unlikely to be removed in the future.

Should we split this up in two separate notes?
- one note listing all currently supported formats (maybe with `.mjs` and `.js` appearing first)
- one not listing all formats that are potential candidates for deprecation

Reply:
I've reworked the section in https://github.com/stylelint/stylelint/pull/8237/commits/58b0544653c4910d61af631529da72ade5952fb9 to:

- frontload all the bits relevant to the recommended route
- clarify that the note relates to formats and locations (and users should probably avoid using either)

Is this clearer, or do we still need to split the notes? I found it tricky to split formats and locations (e.g., filenames, and the `package.json` property) in a way that made sense.

Reply:
I agree :)
This version looks good to me, thank you!